internal/server/server.go

```diff
@@ -196,6 +208,66 @@ func (h *Handlers) GetComposeStatus(ctx echo.Context, composeId string) error {
 	return ctx.JSON(http.StatusOK, status)
 }
 
+func (h *Handlers) GetComposes(ctx echo.Context, params GetComposesParams) error {
+	spec, err := GetSwagger()
+	if err != nil {
+		return err
+	}
+
+	ih := ctx.Get("IdentityHeader")
+	if ih == nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, "Identity Header missing in request handler")
+	}
+	idHeader, ok := ih.(IdentityHeader)
+	if !ok {
+		return echo.NewHTTPError(http.StatusInternalServerError, "Identity Header invalid in request handler")
+	}
+
+	limit := 100
+	if params.Limit != nil {
+		if *params.Limit > 0 {
+			limit = *params.Limit
+		}
+	}
+
+	offset := 0
+	if params.Offset != nil {
+		offset = *params.Offset
+	}
+
+	composes, err := h.server.db.GetComposes(idHeader.Identity.AccountNumber, limit, offset)
+	if err != nil {
+		return err
+	}
+
+	var data []ComposesResponseItem
+	for _, c := range composes {
+		data = append(data, ComposesResponseItem{
+			CreatedAt: c.CreatedAt.String(),
+			Id:        c.Id.String(),
+			Request:   c.Request,
+		})
+	}
+
+	return ctx.JSON(http.StatusOK, ComposesResponse{
+		Meta: struct {
+			Count int `json:"count"`
+		}{
+			len(composes),
+		},
+		Links: struct {
+			First string `json:"first"`
+			Last  string `json:"last"`
+		}{
+			fmt.Sprintf("%v/v%v/composes?offset=%v&limit=%v",
+				RoutePrefix(), spec.Info.Version, offset, limit),
+			fmt.Sprintf("%v/v%v/composes?offset=%v&limit=%v",
+				RoutePrefix(), spec.Info.Version, len(composes)-1, limit),
```

Comment:
What exactly should the semantics of the First and Last links be? In particular, should the `offset` be rounded down to start on the last `limit` interval?

Reply:
I think we have some freedom here, but this was mostly a copy and paste from the packages endpoint. I'm not entirely sure what's more correct, I guess offset = lastOffset+limit is more correct.